chapter_reinforcement_learning/mdp.md

```diff
@@ -0,0 +1,67 @@
+$$
+    R(\tau) = r_0 + r_1 + r_2 + \cdots.
+$$
+
+**The goal in reinforcement learning is to find a trajectory that has the largest return.**
+
+Think of the situation when the robot continues to travel in the gridworld without ever reaching goal location. The sequence of states and actions in a trajectory can be infinitely long in this case and the return of any such infinitely long trajectory will be infinite. In order to keep the RL formulation meaningful even for such trajectories, we introduce the notion of a discount factor $\gamma < 1$. We write the discounted return as
+$$
+    R(\tau) = r_0 + \gamma r_1 + \gamma^2 r_2 + \cdots = \sum_{t=0}^\infty \gamma^t r_t.
+$$
+Note that if $\gamma$ is very small, the rewards earned by the robot in the far future, say $t = 1000$, are heavily discounted by the factor $\gamma^{1000}$. This encourages the robot to select short trajectories that achieve its goal, namely that of going to the green house in the gridwold example (see :numref:`fig_mdp`). For large values of the discount factor, say $\gamma = 0.99$, the robot is encouraged to explore and then find the best trajectory to go to the goal location instead of simply rushing out of the current location.
+
+### Markov assumption in an MDP
+
+We next study the transition function $T(s, a, s') = P(s' \mid s, a)$ in greater detail. Let us think of the state $s_t$ as the location of the robot, as above. And think of the action of the robot as the acceleration that it applies to its wheels instead of an abstract command like "go forward". If the robot has some non-zero velocity at state $s_t$, then the next location $s_{t+1}$ is a function of the past location $s_t$, the acceleration $a_t$ but also the velocity of the robot at time $t$ which is proportional to $s_t - s_{t-1}$. This indicates that we should have
+$$
+    s_{t+1} = \text{function}(s_t, a_t, s_{t-1})
+$$
+which is quite different from our transition function that simply depends upon $s_t$ and $a_t$.
+
+Markov systems are all systems where the next state $s_{t+1}$ is only a function of the current $s_t$ and the action taken at the current state $a_t$; the next state does not depend on which actions were taken in the past or the states that the robot was at in the past. Assuming that the transition function is Markovian, for example the system above is not Markovian because the next location $s_{t+1}$ depends upon the previous state $s_{t-1}$ through the velocity. However this is not a restrictive assumption. MDPs are still capable of modeling a very large class of interesting systems. For example, in the system above, if we chose our state $s_t$ to the tuple $\text{location, velocity}$ then the system above is Markovian because its next state depends only upon the current state and the action at the current state.
```

Comment:
`*Markov systems*` (highlight terms in definitions)

Reply:
the action taken at the current state $a_t$
->
the action taken $a_t$ at the current state